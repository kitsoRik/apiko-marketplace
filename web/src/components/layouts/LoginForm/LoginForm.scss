.login-container {
    width: 425px;

    position: relative;

    display: grid;
    grid-template-rows: auto 75px;
    grid-row-gap: 20px;
}

@keyframes rotating {
    from {
        transform: rotate(0);
    }

    to {
        transform: rotate(360deg);
    }
}