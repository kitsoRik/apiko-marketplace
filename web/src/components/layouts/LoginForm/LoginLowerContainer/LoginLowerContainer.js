import React from 'react';

import './LoginLowerContainer.scss';

const LoginLowerContainer = ({ children }) => {
    return ( 
        <div className="login-lower-container">
            { children }
        </div>
     );
}

export default LoginLowerContainer;