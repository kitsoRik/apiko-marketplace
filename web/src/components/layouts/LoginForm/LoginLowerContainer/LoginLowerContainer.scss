.login-lower-container {
    width: 100%;

    display: flex;
    justify-content: center;
    align-items: center;

    box-shadow: 0px 2px 42px rgba(0, 0, 0, 0.111233);
}