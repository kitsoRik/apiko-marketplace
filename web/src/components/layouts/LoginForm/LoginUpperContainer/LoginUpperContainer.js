import React from "react";

import "./LoginUpperContainer.scss";

const LoginUpperContainer = ({ children }) => {
	return <div className="login-upper-container">{children}</div>;
};

export default LoginUpperContainer;
