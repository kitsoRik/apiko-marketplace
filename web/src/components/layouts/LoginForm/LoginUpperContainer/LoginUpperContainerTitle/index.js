import LoginUpperContainerTitle from "./LoginUpperContainerTitle";

export default LoginUpperContainerTitle;
