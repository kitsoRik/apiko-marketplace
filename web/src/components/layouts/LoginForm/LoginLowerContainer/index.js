import LoginLowerContainer from "./LoginLowerContainer";

export default LoginLowerContainer;