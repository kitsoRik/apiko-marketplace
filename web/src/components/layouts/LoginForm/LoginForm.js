import React from 'react';
import LoginUpperContainer from './LoginUpperContainer/LoginUpperContainer';
import LoginLowerContainer from './LoginLowerContainer/LoginLowerContainer';

import "./LoginForm.scss";
import ModalLoading from '../ModalLoading/ModalLoading';

const LoginForm = ({ className, loading, children }) => {
    const ch = Array.isArray(children) ? children : [children];

    const upper = ch.find(c => c && c.type === LoginUpperContainer );
    const lower = ch.find(c => c && c.type === LoginLowerContainer );

    return ( 
        <div 
            className={`login-container ${className}`}
            style={{ gridTemplateRows: `${upper ? 'auto' : ''} ${lower ? '75px' : ''}`}}>
            { upper }
            { lower }
            { loading && <ModalLoading />}
        </div>
     );
}

export default LoginForm;