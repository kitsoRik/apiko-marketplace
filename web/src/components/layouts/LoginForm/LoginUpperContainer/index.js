import LoginUpperContainer from "./LoginUpperContainer";

export default LoginUpperContainer;
