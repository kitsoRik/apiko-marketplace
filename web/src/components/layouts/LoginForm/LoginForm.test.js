import React from "react";
import { create } from "react-test-renderer";
import LoginForm from "./LoginForm";
import LoginUpperContainer from "./LoginUpperContainer/LoginUpperContainer";
import LoginLowerContainer from "./LoginLowerContainer/LoginLowerContainer";

describe("LoginForm", () => {
	it("should be rendered", () => {
		const component = create(<LoginForm></LoginForm>);
		const json = component.toJSON();

		expect(json).toMatchSnapshot();
	});
	it("should render with upper container", () => {
		const component = create(
			<LoginForm>
				<LoginUpperContainer>...</LoginUpperContainer>
			</LoginForm>
		);
		const json = component.toJSON();

		expect(json).toMatchSnapshot();
	});
	it("should render with lower container", () => {
		const component = create(
			<LoginForm>
				<LoginLowerContainer>...</LoginLowerContainer>
			</LoginForm>
		);
		const json = component.toJSON();

		expect(json).toMatchSnapshot();
	});
	it("should render with both containers", () => {
		const component = create(
			<LoginForm>
				<LoginUpperContainer>...</LoginUpperContainer>
				<LoginLowerContainer>...</LoginLowerContainer>
			</LoginForm>
		);
		const json = component.toJSON();

		expect(json).toMatchSnapshot();
	});
	it("should render the loading", () => {
		const component = create(<LoginForm loading={true}></LoginForm>);
		const json = component.toJSON();

		expect(json).toMatchSnapshot();
	});
	it("should render the loading with containers", () => {
		let component = create(
			<LoginForm loading={true}>
				<LoginUpperContainer>...</LoginUpperContainer>
			</LoginForm>
		);
		let json = component.toJSON();

		expect(json).toMatchSnapshot();

		component = create(
			<LoginForm loading={true}>
				<LoginLowerContainer>...</LoginLowerContainer>
			</LoginForm>
		);
		json = component.toJSON();

		expect(json).toMatchSnapshot();

		component = create(
			<LoginForm loading={true}>
				<LoginUpperContainer>...</LoginUpperContainer>
				<LoginLowerContainer>...</LoginLowerContainer>
			</LoginForm>
		);
		json = component.toJSON();

		expect(json).toMatchSnapshot();
	});
});
