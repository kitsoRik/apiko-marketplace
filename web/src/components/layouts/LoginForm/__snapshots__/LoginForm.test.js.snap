// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginForm should be rendered 1`] = `
<form
  className="login-container undefined"
  style={
    Object {
      "gridTemplateRows": " ",
    }
  }
/>
`;

exports[`LoginForm should render the loading 1`] = `
<form
  className="login-container undefined"
  style={
    Object {
      "gridTemplateRows": " ",
    }
  }
>
  <div
    className="modal-loading undefined"
    darken="true"
    onClick={[Function]}
  >
    <img
      alt="Loading"
      src="loading.svg"
      style={
        Object {
          "maxHeight": "100%",
          "maxWidth": "100%",
        }
      }
    />
  </div>
</form>
`;

exports[`LoginForm should render the loading with containers 1`] = `
<form
  className="login-container undefined"
  style={
    Object {
      "gridTemplateRows": "auto ",
    }
  }
>
  <div
    className="login-upper-container"
  >
    ...
  </div>
  <div
    className="modal-loading undefined"
    darken="true"
    onClick={[Function]}
  >
    <img
      alt="Loading"
      src="loading.svg"
      style={
        Object {
          "maxHeight": "100%",
          "maxWidth": "100%",
        }
      }
    />
  </div>
</form>
`;

exports[`LoginForm should render the loading with containers 2`] = `
<form
  className="login-container undefined"
  style={
    Object {
      "gridTemplateRows": " 75px",
    }
  }
>
  <div
    className="login-lower-container"
  >
    ...
  </div>
  <div
    className="modal-loading undefined"
    darken="true"
    onClick={[Function]}
  >
    <img
      alt="Loading"
      src="loading.svg"
      style={
        Object {
          "maxHeight": "100%",
          "maxWidth": "100%",
        }
      }
    />
  </div>
</form>
`;

exports[`LoginForm should render the loading with containers 3`] = `
<form
  className="login-container undefined"
  style={
    Object {
      "gridTemplateRows": "auto 75px",
    }
  }
>
  <div
    className="login-upper-container"
  >
    ...
  </div>
  <div
    className="login-lower-container"
  >
    ...
  </div>
  <div
    className="modal-loading undefined"
    darken="true"
    onClick={[Function]}
  >
    <img
      alt="Loading"
      src="loading.svg"
      style={
        Object {
          "maxHeight": "100%",
          "maxWidth": "100%",
        }
      }
    />
  </div>
</form>
`;

exports[`LoginForm should render with both containers 1`] = `
<form
  className="login-container undefined"
  style={
    Object {
      "gridTemplateRows": "auto 75px",
    }
  }
>
  <div
    className="login-upper-container"
  >
    ...
  </div>
  <div
    className="login-lower-container"
  >
    ...
  </div>
</form>
`;

exports[`LoginForm should render with lower container 1`] = `
<form
  className="login-container undefined"
  style={
    Object {
      "gridTemplateRows": " 75px",
    }
  }
>
  <div
    className="login-lower-container"
  >
    ...
  </div>
</form>
`;

exports[`LoginForm should render with upper container 1`] = `
<form
  className="login-container undefined"
  style={
    Object {
      "gridTemplateRows": "auto ",
    }
  }
>
  <div
    className="login-upper-container"
  >
    ...
  </div>
</form>
`;
